File: TinyScript/java/src/main/java/paser/ast/ReturnStmt.java
package paser.ast;

import paser.util.ParseException;
import paser.util.PeekTokenIterator;

/**
 * @author MFine
 * @version 1.0
 * @date 2021/10/28 22:50
 **/
public class ReturnStmt extends Stmt {

    protected ReturnStmt(ASTNode parent) {
        super(parent,ASTNodeTypes.RETURN_STMT,"return");
    }

    public static ASTNode parse(ASTNode parent, PeekTokenIterator it) throws ParseException {
        it.nextMatch("return");

        ReturnStmt returnStmt = new ReturnStmt(parent);
        ASTNode parse = Expr.parse(it);
        returnStmt.addChild(parse);
        return returnStmt;
    }

}

File: TinyScript/js/src/main/js/parser/ast/ReturnStmt.ts
import Stmt from "./Stmt";
import ASTNodeTypes from "./ASTNodeTypes";
import PeekTokenIterator from "../PeekTokenIterator";

export  default  class ReturnStmt extends Stmt {
    constructor() {
        super(ASTNodeTypes.RETURN_STMT, "return");
    }

    static parse(it:PeekTokenIterator):ReturnStmt{
       const {Expr}  = require('./index')
        it.nextMatch("return");
        const stmt = new ReturnStmt();
        const parse = Expr.parse(it);
        stmt.addChild(parse);
        return stmt;
    }
}